src/anm/utils.js

```diff
@@ -43,6 +43,41 @@ is.tween = function(f) {
     return f.is_tween && is.modifier(f);
 };
 
+is.equal = function(x, y) {
+    if (x === y) return true;
+    // if both x and y are null or undefined and exactly the same
+
+    if (!(x instanceof Object) || !(y instanceof Object)) return false;
+    // if they are not strictly equal, they both need to be Objects
+
+    if (x.constructor !== y.constructor) return false;
+    // they must have the exact same prototype chain, the closest we can do is
+    // test their constructor.
+
+    for (var p in x) {
```

Comment:
Agree with _not a big deal_. But actually I try to fallback to ID-comparison everywhere where it's needed just because I don't really like deep-object analysis :). Object-equality is really not so often-required operation, as for me. For Elements or Modifiers/Painters, where it's needed in several cases, ID's are enough, and do work well.